File: 202. Happy Number.py
class Solution:
    def isHappy(self, n: int) -> bool:
        ls = []
        while n not in ls:
            ls.append(n)
            res = 0
            while n :
                n, mod = divmod(n, 10)
                res += mod**2
            if res == 1:
                return True
            n = res
        return False

File: 202. Happy Number.java
class Solution {
    public boolean isHappy(int n) {
        HashSet<Integer> seen = new HashSet<>();
        while (n != 1) {
            seen.add(n);
            int curr = 0;
            while (n != 0) {
                curr += (n%10) * (n%10);
                n = n/10;
            }
            n = curr;
            if (seen.contains(n)) {
                return false;
            }
        }
        return true;
    }
}